src/core/symbolic.cc

```diff
@@ -10,7 +10,7 @@
 namespace nnvm {
 
 namespace symbol_constants {
-const char *kNamespaceSeparator = "_";
+const char *kNamespaceSeparator = "$";
```

Comment:
As long as the front-end behavior is fixed, I am fine with this change. Consider use "/" instead for separator, which is more typical